[BUG] Class Not Found Exception

**Describe the bug**
Using the API v0.2.3 Snapshot we're getting the error that classes are not being found (even though the module was loaded)
```
Caused by: java.lang.ClassNotFoundException: xyz.theprogramsrc.tasksmodule.spigot.SpigotTasks
at java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[?:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
```

**To Reproduce**
Steps to reproduce the behavior:
1. Install v0.2.3 of SimpleCoreAPI
2. Install the latest SimpleCore-UIsModule build
3. Write in chat `simpleui`
4. See error

**Expected behavior**
UI should be opened and error shouldn't be shown

**System Details (please complete the following information):**
 - OS: Paper
 - OS Version: 1.18
 - Version: v0.2.3 Snapshot

**Additional context**
I believe this is an issue caused by the module loader not loading classes as expected. I think maybe include the class loader from the BungeeLoader or SpigotLoader as argument may help. Also I'll try checking the dependency downloader module, that works and may have some answers.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## v0.2.4 - Snapshot
+* Fixed Class Not Found error
+* Improvements to the Module Loader
+
 ## v0.2.3 - Snapshot
 * Fixes to update checker to print the message
 * Now GitHub actions will automatically upload the jar file

--- a/build.gradle
+++ b/build.gradle
@@ -7,7 +7,7 @@ plugins {
     id 'org.jetbrains.dokka' version '1.6.0'
 }
 
-def projectVersion = (System.getenv("VERSION") ?: '0.2.3-SNAPSHOT').replaceFirst("v", "").replace('/', '')
+def projectVersion = (System.getenv("VERSION") ?: '0.2.4-SNAPSHOT').replaceFirst("v", "").replace('/', '')
 
 group 'xyz.theprogramsrc'
 version projectVersion

--- a/src/main/kotlin/xyz/theprogramsrc/simplecoreapi/global/module/ModuleManager.kt
+++ b/src/main/kotlin/xyz/theprogramsrc/simplecoreapi/global/module/ModuleManager.kt
@@ -126,11 +126,13 @@ class ModuleManager(private val logger: Logger) {
             moduleDependencies[it.key] = it.value.second.dependencies.toList()
         }
 
+        val urlClassLoader = URLClassLoader(modules.map { it.value.first }.map { it.toURI().toURL() }.toTypedArray(), this::class.java.classLoader)
+
         ModuleHelper.sortModuleDependencies(moduleDependencies).forEach { moduleName ->
             if(!loadedModules.contains(moduleName) && modules.containsKey(moduleName)) {
                 modules[moduleName]?.let { pair ->
                     try {
-                        loadIntoClasspath(pair.first, pair.second)
+                        loadIntoClasspath(urlClassLoader, pair.first, pair.second)
                         loadedModules.add(moduleName)
                     } catch (e: InvalidModuleException) {
                         e.printStackTrace()
@@ -145,40 +147,38 @@ class ModuleManager(private val logger: Logger) {
 
     /**
      * Loads a module into the classpath
+     * @param loader The [URLClassLoader] to load the module into
      * @param file The file to load
      * @param description The module description
      * @throws InvalidModuleException If the main module class is invalid
      * @throws ModuleLoadException If the module failed to load
      */
     @Throws(InvalidModuleException::class, ModuleLoadException::class)
-    private fun loadIntoClasspath(file: File, description: ModuleDescription) {
+    private fun loadIntoClasspath(loader: URLClassLoader, file: File, description: ModuleDescription) {
         var entry: ZipEntry?
         try {
-            URLClassLoader(arrayOf(file.toURI().toURL()), this.javaClass.classLoader).use { loader ->
-                JarInputStream(FileInputStream(file)).use { jarInputStream ->
-                    while (jarInputStream.nextEntry.also { entry = it } != null) {
-                        val entryName = entry?.name ?: continue
-                        if (entryName.endsWith(".class")) {
-                            val name = entryName.replace("/", ".").replace(".class", "")
-                            val clazz = loader.loadClass(name)
-                            if (name == description.mainClass) {
-                                if (!Module::class.java.isAssignableFrom(clazz)) {
-                                    throw InvalidModuleException("The class ${description.mainClass} must be extended to the Module class!")
-                                }
-                                try {
-                                    logger.info("Loading module ${description.name} v${description.version}")
-                                    val moduleClass = clazz.asSubclass(Module::class.java)
-                                    val module = moduleClass.getConstructor().newInstance()
-                                    module.init(file, description)
-                                    module.onEnable()
-                                    loadedModules[description.name] = module
-                                    logger.info("Module ${description.name} v${description.version} loaded!")
-                                } catch (e: Exception) {
-                                    throw ModuleLoadException("Failed to load module ${description.name} v${description.version}", e)
-                                }
-                            }
-                        }
-                    }
+            JarInputStream(FileInputStream(file)).use { jarInputStream ->
+                val mainClass = loader.loadClass(description.mainClass)
+                if(!Module::class.java.isAssignableFrom(mainClass)){
+                    throw InvalidModuleException("The class ${description.mainClass} must be extended to the Module class!")
+                }
+
+                while (jarInputStream.nextEntry.also { entry = it } != null) {
+                    val entryName = entry?.name ?: continue
+                    if(!entryName.endsWith(".class")) continue
+                    loader.loadClass(entryName.replace('/', '.').replace(".class", ""))
+                }
+
+                try {
+                    logger.info("Loading module ${description.name} v${description.version}")
+                    val moduleClass = mainClass.asSubclass(Module::class.java)
+                    val module = moduleClass.getConstructor().newInstance()
+                    module.init(file, description)
+                    module.onEnable()
+                    loadedModules[description.name] = module
+                    logger.info("Module ${description.name} v${description.version} loaded!")
+                } catch (e: Exception) {
+                    throw ModuleLoadException("Failed to load module ${description.name} v${description.version}", e)
                 }
             }
         } catch (e: Exception) {

